staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -72,10 +74,12 @@ func newETCD3HealthCheck(c storagebackend.Config) (func() error, error) {
 		client := clientValue.Load().(*clientv3.Client)
 		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
 		defer cancel()
-		if _, err := client.Cluster.MemberList(ctx); err != nil {
```

Comment:
@lorenz etcd folks are running the client.Get on each individual endpoint i believe

See endpointsFromCluster where they run MemberList and then subsequently do the fake get.

https://github.com/etcd-io/etcd/blob/master/etcdctl/ctlv3/command/ep_command.go#L97-L121

Reply:
Correct, but as far as I know kube-apiserver only ever connects to one etcd server, right? So it doesn't matter if the cluster is healthy, just the instance it connects to.